android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -257,5 +279,18 @@ private TokenDetails readTokenDetails(Map<String, Object> jsonMap) {
         return jsonMap;
     }
 
+    private Map<String, Object> encodeChannelMessage(Message c){
+        if(c==null) return null;
```

Comment:
Would you like to have an issue for this?
I can only think about moving this null check to `encodeAblyMessage` in `Codec.dart` once we implement all data types that required to be encoded.

Reply:
@QuintinWillison Would you like to have an issue for this?
I can only think about moving this null check to `encodeAblyMessage` in `Codec.dart` once we implement all data types that required to be encoded.

Reply:
@zoechi am I fussing over nothing here? I would value your opinion on this. Thanks!

Reply:
`null` is a valid JSON value and if the message passed is `null`, then `null` is the correct conversion result.
So, I do not see anything wrong with this.

It would be different if `null` were not a valid value for the parameter `c`, then an assert or throwing an exception would be more appropriate.

Reply:
Back in my box. 😝